pkg/customunit/handler.go

```diff
@@ -391,10 +394,37 @@ func (c *CustomUnitHandler) ManagedApplicationLookup(appLookup *customunits.AppL
 		ID:            managedApp.Metadata.ID,
 		Name:          managedApp.Name,
 		ConsumerOrgID: consumerOrgID,
-	}, nil
+	}
+}
+
+func (c *CustomUnitHandler) customAppLookup(appValue, appCustomAttr string) *v1.ResourceInstance {
+	if appCustomAttr == "" {
+		c.logger.Error("not able to lookup application by custom attribute without the attribute name set")
+		return nil
+	}
+
+	customKey := fmt.Sprintf("%s_%s", appCustomAttr, appValue)
+	if apiID, ok := c.apiCustomLookups[customKey]; ok {
+		return c.cache.GetAPIServiceWithAPIID(apiID)
+	}
+
+	for _, key := range c.cache.GetAPIServiceKeys() {
```

Comment:
Should this be GetManagedApplicationCacheKeys() ?

Reply:
yessir